Use AtomString instead of AtomStringImpl in TreeScopeOrderedMap

rniwa opened the pull request:
#### f0939c55f2d773fbe9666e902d385acd05100e48
<pre>
Use AtomString instead of AtomStringImpl in TreeScopeOrderedMap
<a href="https://bugs.webkit.org/show_bug.cgi?id=263239">https://bugs.webkit.org/show_bug.cgi?id=263239</a>

Reviewed by Chris Dumez.

Replaced the use of AtomStringImpl* / AtomStringImpl&amp; in TreeScopeOrderedMap with AtomString.

* Source/WebCore/bindings/js/JSDOMWindowProperties.cpp:
(WebCore::jsDOMWindowPropertiesGetOwnPropertySlotNamedItemGetter):
* Source/WebCore/dom/Element.cpp:
(WebCore::Element::insertedIntoAncestor):
(WebCore::Element::removedFromAncestor):
(WebCore::Element::updateNameForTreeScope):
(WebCore::Element::updateNameForDocument):
(WebCore::Element::updateIdForTreeScope):
(WebCore::Element::updateIdForDocument):
(WebCore::Element::updateLabel):
* Source/WebCore/dom/ImageOverlay.cpp:
(WebCore::ImageOverlay::hasOverlay):
* Source/WebCore/dom/TreeScope.cpp:
(WebCore::TreeScope::getElementById const):
(WebCore::TreeScope::getAllElementsById const):
(WebCore::TreeScope::addElementById):
(WebCore::TreeScope::removeElementById):
(WebCore::TreeScope::getElementByName const):
(WebCore::TreeScope::addElementByName):
(WebCore::TreeScope::removeElementByName):
(WebCore::TreeScope::addImageMap):
(WebCore::TreeScope::removeImageMap):
(WebCore::TreeScope::getImageMap const):
(WebCore::TreeScope::addImageElementByUsemap):
(WebCore::TreeScope::removeImageElementByUsemap):
(WebCore::TreeScope::imageElementByUsemap const):
(WebCore::TreeScope::addLabel):
(WebCore::TreeScope::removeLabel):
(WebCore::TreeScope::labelElementsForId):
* Source/WebCore/dom/TreeScope.h:
* Source/WebCore/dom/TreeScopeInlines.h:
(WebCore::TreeScope::hasElementWithId const):
(WebCore::TreeScope::containsMultipleElementsWithId const):
(WebCore::TreeScope::hasElementWithName const):
(WebCore::TreeScope::containsMultipleElementsWithName const):
* Source/WebCore/dom/TreeScopeOrderedMap.cpp:
(WebCore::TreeScopeOrderedMap::add):
(WebCore::TreeScopeOrderedMap::remove):
(WebCore::TreeScopeOrderedMap::get const):
(WebCore::TreeScopeOrderedMap::getAll const):
(WebCore::TreeScopeOrderedMap::getElementById const):
(WebCore::TreeScopeOrderedMap::getElementByName const):
(WebCore::TreeScopeOrderedMap::getElementByMapName const):
(WebCore::TreeScopeOrderedMap::getElementByUsemap const):
(WebCore::TreeScopeOrderedMap::getElementsByLabelForAttribute const):
(WebCore::TreeScopeOrderedMap::getElementByWindowNamedItem const):
(WebCore::TreeScopeOrderedMap::getElementByDocumentNamedItem const):
(WebCore::TreeScopeOrderedMap::getAllElementsById const):
(WebCore::TreeScopeOrderedMap::keys const):
* Source/WebCore/dom/TreeScopeOrderedMap.h:
(WebCore::TreeScopeOrderedMap::containsSingle const):
(WebCore::TreeScopeOrderedMap::contains const):
(WebCore::TreeScopeOrderedMap::containsMultiple const):
* Source/WebCore/dom/mac/ImageControlsMac.cpp:
(WebCore::ImageControlsMac::hasImageControls):
* Source/WebCore/html/CachedHTMLCollectionInlines.h:
(WebCore::traversalType&gt;::namedItem const):
* Source/WebCore/html/HTMLDocument.cpp:
(WebCore::HTMLDocument::namedItem):
(WebCore::HTMLDocument::isSupportedPropertyName const):
(WebCore::HTMLDocument::addDocumentNamedItem):
(WebCore::HTMLDocument::removeDocumentNamedItem):
(WebCore::HTMLDocument::addWindowNamedItem):
(WebCore::HTMLDocument::removeWindowNamedItem):
* Source/WebCore/html/HTMLDocument.h:
(WebCore::HTMLDocument::documentNamedItem const):
(WebCore::HTMLDocument::hasDocumentNamedItem const):
(WebCore::HTMLDocument::documentNamedItemContainsMultipleElements const):
(WebCore::HTMLDocument::windowNamedItem const):
(WebCore::HTMLDocument::hasWindowNamedItem const):
(WebCore::HTMLDocument::windowNamedItemContainsMultipleElements const):
* Source/WebCore/html/HTMLImageElement.cpp:
(WebCore::HTMLImageElement::attributeChanged):
(WebCore::HTMLImageElement::insertedIntoAncestor):
(WebCore::HTMLImageElement::removedFromAncestor):
(WebCore::HTMLImageElement::matchesUsemap const):
* Source/WebCore/html/HTMLImageElement.h:
* Source/WebCore/html/HTMLMapElement.cpp:
(WebCore::HTMLMapElement::imageElement):
* Source/WebCore/html/HTMLNameCollection.cpp:
(WebCore::WindowNameCollection::elementMatches):
(WebCore::DocumentNameCollection::elementMatches):
* Source/WebCore/html/HTMLNameCollection.h:
* Source/WebCore/html/HTMLObjectElement.cpp:
(WebCore::HTMLObjectElement::updateExposedState):

Canonical link: <a href="https://commits.webkit.org/269477@main">https://commits.webkit.org/269477@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5165cf5e814bb0e0ada9f25f277083bbc68d27d8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/22593 "4 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/245 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/23677 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/24499 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/20914 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/22852 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/1324 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/23124 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/21881 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/22832 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/162 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/19607 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/25352 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/22776 "Passed tests") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/162 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/20474 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/26717 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/19672 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/20536 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/20718 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/24552 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/21960 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/153 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/18015 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/26010 "Built successfully") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/22388 "Passed tests") | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/121 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/20277 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/6103 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/173 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/27290 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/2855 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/169 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/5929 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

cdumez reviewed (approved):
LGTM. I hope this doesn't regress perf.

rniwa committed:
Use AtomString instead of AtomStringImpl in TreeScopeOrderedMap
https://bugs.webkit.org/show_bug.cgi?id=263239

Reviewed by Chris Dumez.

Replaced the use of AtomStringImpl* / AtomStringImpl& in TreeScopeOrderedMap with AtomString.

* Source/WebCore/bindings/js/JSDOMWindowProperties.cpp:
(WebCore::jsDOMWindowPropertiesGetOwnPropertySlotNamedItemGetter):
* Source/WebCore/dom/Element.cpp:
(WebCore::Element::insertedIntoAncestor):
(WebCore::Element::removedFromAncestor):
(WebCore::Element::updateNameForTreeScope):
(WebCore::Element::updateNameForDocument):
(WebCore::Element::updateIdForTreeScope):
(WebCore::Element::updateIdForDocument):
(WebCore::Element::updateLabel):
* Source/WebCore/dom/ImageOverlay.cpp:
(WebCore::ImageOverlay::hasOverlay):
* Source/WebCore/dom/TreeScope.cpp:
(WebCore::TreeScope::getElementById const):
(WebCore::TreeScope::getAllElementsById const):
(WebCore::TreeScope::addElementById):
(WebCore::TreeScope::removeElementById):
(WebCore::TreeScope::getElementByName const):
(WebCore::TreeScope::addElementByName):
(WebCore::TreeScope::removeElementByName):
(WebCore::TreeScope::addImageMap):
(WebCore::TreeScope::removeImageMap):
(WebCore::TreeScope::getImageMap const):
(WebCore::TreeScope::addImageElementByUsemap):
(WebCore::TreeScope::removeImageElementByUsemap):
(WebCore::TreeScope::imageElementByUsemap const):
(WebCore::TreeScope::addLabel):
(WebCore::TreeScope::removeLabel):
(WebCore::TreeScope::labelElementsForId):
* Source/WebCore/dom/TreeScope.h:
* Source/WebCore/dom/TreeScopeInlines.h:
(WebCore::TreeScope::hasElementWithId const):
(WebCore::TreeScope::containsMultipleElementsWithId const):
(WebCore::TreeScope::hasElementWithName const):
(WebCore::TreeScope::containsMultipleElementsWithName const):
* Source/WebCore/dom/TreeScopeOrderedMap.cpp:
(WebCore::TreeScopeOrderedMap::add):
(WebCore::TreeScopeOrderedMap::remove):
(WebCore::TreeScopeOrderedMap::get const):
(WebCore::TreeScopeOrderedMap::getAll const):
(WebCore::TreeScopeOrderedMap::getElementById const):
(WebCore::TreeScopeOrderedMap::getElementByName const):
(WebCore::TreeScopeOrderedMap::getElementByMapName const):
(WebCore::TreeScopeOrderedMap::getElementByUsemap const):
(WebCore::TreeScopeOrderedMap::getElementsByLabelForAttribute const):
(WebCore::TreeScopeOrderedMap::getElementByWindowNamedItem const):
(WebCore::TreeScopeOrderedMap::getElementByDocumentNamedItem const):
(WebCore::TreeScopeOrderedMap::getAllElementsById const):
(WebCore::TreeScopeOrderedMap::keys const):
* Source/WebCore/dom/TreeScopeOrderedMap.h:
(WebCore::TreeScopeOrderedMap::containsSingle const):
(WebCore::TreeScopeOrderedMap::contains const):
(WebCore::TreeScopeOrderedMap::containsMultiple const):
* Source/WebCore/dom/mac/ImageControlsMac.cpp:
(WebCore::ImageControlsMac::hasImageControls):
* Source/WebCore/html/CachedHTMLCollectionInlines.h:
(WebCore::traversalType>::namedItem const):
* Source/WebCore/html/HTMLDocument.cpp:
(WebCore::HTMLDocument::namedItem):
(WebCore::HTMLDocument::isSupportedPropertyName const):
(WebCore::HTMLDocument::addDocumentNamedItem):
(WebCore::HTMLDocument::removeDocumentNamedItem):
(WebCore::HTMLDocument::addWindowNamedItem):
(WebCore::HTMLDocument::removeWindowNamedItem):
* Source/WebCore/html/HTMLDocument.h:
(WebCore::HTMLDocument::documentNamedItem const):
(WebCore::HTMLDocument::hasDocumentNamedItem const):
(WebCore::HTMLDocument::documentNamedItemContainsMultipleElements const):
(WebCore::HTMLDocument::windowNamedItem const):
(WebCore::HTMLDocument::hasWindowNamedItem const):
(WebCore::HTMLDocument::windowNamedItemContainsMultipleElements const):
* Source/WebCore/html/HTMLImageElement.cpp:
(WebCore::HTMLImageElement::attributeChanged):
(WebCore::HTMLImageElement::insertedIntoAncestor):
(WebCore::HTMLImageElement::removedFromAncestor):
(WebCore::HTMLImageElement::matchesUsemap const):
* Source/WebCore/html/HTMLImageElement.h:
* Source/WebCore/html/HTMLMapElement.cpp:
(WebCore::HTMLMapElement::imageElement):
* Source/WebCore/html/HTMLNameCollection.cpp:
(WebCore::WindowNameCollection::elementMatches):
(WebCore::DocumentNameCollection::elementMatches):
* Source/WebCore/html/HTMLNameCollection.h:
* Source/WebCore/html/HTMLObjectElement.cpp:
(WebCore::HTMLObjectElement::updateExposedState):

Canonical link: https://commits.webkit.org/269477@main
